From 916213b1bfcdeaec158d3edfb9104d199170cb20 Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Thu, 26 Oct 2017 21:29:27 -0700
Subject: [PATCH 103/105] gpio: sunxi: Don't use simple_strtol

We're pretty close to SPL size limit on 64-bit sunxi, so
save some memory but not using simple_strtol which is 152 bytes

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 drivers/gpio/sunxi_gpio.c | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/drivers/gpio/sunxi_gpio.c b/drivers/gpio/sunxi_gpio.c
index 7be9e25cd7..0c93cadaa8 100644
--- a/drivers/gpio/sunxi_gpio.c
+++ b/drivers/gpio/sunxi_gpio.c
@@ -101,8 +101,7 @@ int sunxi_name_to_gpio(const char *name)
 {
 	int group = 0;
 	int groupsize = 9 * 32;
-	long pin;
-	char *eptr;
+	long pin = 0;
 
 	if (*name == 'P' || *name == 'p')
 		name++;
@@ -112,9 +111,16 @@ int sunxi_name_to_gpio(const char *name)
 		name++;
 	}
 
-	pin = simple_strtol(name, &eptr, 10);
-	if (!*name || *eptr)
-		return -1;
+	for (;;) {
+		if (*name == 0)
+			break;
+		if (*name < '0' || *name > '9')
+			return -1;
+		pin *= 10;
+		pin += *name - '0';
+		name++;
+	}
+
 	if (pin < 0 || pin > groupsize || group >= 9)
 		return -1;
 	return group * 32 + pin;
-- 
2.13.6

